presto-main/src/main/java/com/facebook/presto/cost/CachingCostProvider.java

```diff
@@ -95,7 +102,7 @@ private PlanNodeCostEstimate calculateCumulativeCost(PlanNode node)
     {
         PlanNodeCostEstimate localCosts = costCalculator.calculateCost(node, statsProvider, lookup, session, types);
 
-        PlanNodeCostEstimate sourcesCost = node.getSources().stream()
+        PlanNodeCostEstimate sourcesCost = sourceProvider.getSources(node).stream()
```

Comment:
Can sourceProvider be modelled with a lookup?

Reply:
not without abusing the interface :).  Lookups are for resolving GroupReferences to the nodes that are a part of that group.  A SourceProvider gives you the list of source nodes for a given node.